File: Labs/Lab 2/C++/StringCalculator/DigitString.cpp
#include "DigitString.h"
#include <algorithm>

//Конструктор за замовчуванням
DigitString::DigitString()
    : value("")
{

}

//Конструктор з параметром
DigitString::DigitString(const std::string& initialValue)
    : value(initialValue)
{

}

//Конструктор копіювання
DigitString::DigitString(const DigitString& other)
    : value(other.value)
{

}

//Деструктор
DigitString::~DigitString()
{

}

int DigitString::getLength() const
{
    return static_cast<int>(value.size());
}

void DigitString::removeChar5()
{
    value.erase(std::remove(value.begin(), value.end(), '5'), value.end());
}

// Повернути поточний рядок
std::string DigitString::getValue() const
{
    return value;
}

File: Labs/Lab 2/C#/StringCalculator/DigitString.cs
using System;

namespace DigitStringLib
{
    public class DigitString
    {
        private string value;

        //Конструктор за замовчуванням
        public DigitString()
        {
            value = "";
        }

        //Конструктор з параметром
        public DigitString(string initialValue)
        {
            value = initialValue;
        }

        //Конструктор копіювання
        public DigitString(DigitString other)
        {
            this.value = other.value;
        }

        //Фіналізатор
        ~DigitString()
        {

        }

        public int GetLength()
        {
            return value.Length;
        }


        public void RemoveChar5()
        {
            value = value.Replace("5", "");
        }

        public string GetValue()
        {
            return value;
        }
    }
}